WordPress/gutenberg
# Update has-text has-icon button spacing

## What?
Updates the padding applied the `has-text has-icon` buttons. Also reduces the spacing between label + icon, and uses `gap` rather than `margin`.

## Why?
The current spacing feels a but unbalanced. `gap` works better for RTL compared to left/right margin.

## How?
CSS.

## Testing Instructions
Observe equal left/right padding applied to `has-text has-icon` like the View button in the Top Bar.

## Comparison

| Before | After |
| --- | --- |
| <img width="100" alt="Screenshot 2023-05-03 at 09 18 21" src="https://user-images.githubusercontent.com/846565/235865654-17755bb5-d1ea-407f-b780-14d95334311a.png"> | <img width="100" alt="Screenshot 2023-05-03 at 09 26 08" src="https://user-images.githubusercontent.com/846565/235866358-f9d97aa2-33e7-490f-9049-2ca463406aa6.png"> |

Note: The outline in the screenshot above is for illustrative purposes only. Here's how the button actually appears:

![button](https://user-images.githubusercontent.com/846565/235866725-3a4b1bcf-e034-4114-b9ff-96a30edef06e.gif)
